apps/devhub/tests/test_views.py

```diff
@@ -1165,6 +1168,8 @@ def test_step_2_with_cookie(self):
 
     def test_step_2_no_cookie(self):
         # We require a cookie that gets set in step 1.
+        self.user.update(read_dev_agreement=None)
+
         r = self.client.get(reverse('devhub.submit.2'), follow=True)
         self.assertRedirects(r, reverse('devhub.submit.1'))
```

Comment:
Not exactly. Those tests are checking that a developer can post (to say they've read the agreement) and then continue. This does not cover the behavior of a developer who has logged out, gone off, then returned to the devhub site later. What you need is to initialize their user with `read_dev_agreement=now()`, load the first view with a GET (not a post), and make sure it is redirected.